### PR TITLE
GHC-1629

### PR DESCRIPTION
**Pull request by fisx**
Please be harsh with your review, i want the next PR to be better!  :)

### CODE REVIEW

File: message-index/messages/GHC-1629/index.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: Too many type arguments in constructor pattern
+summary: You have applied to many types to a constructor in a pattern.
```
Comment (david-christiansen) [inline message-index/messages/GHC-1629/index.md:3]:
```suggestion
summary: You have applied too many types to a constructor in a pattern.
```

File: message-index/messages/GHC-1629/example1/index.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+title: Too many type arguments in constructor pattern
+---
+
```
Comment (PPKFS) [inline message-index/messages/GHC-1629/example1/index.md:4]: An explanation of this specific instance of the error would be useful (i.e. the @Bool is unnecessary)

File: message-index/messages/GHC-1629/index.md (hunk 1 above)
Comment (PPKFS) [inline message-index/messages/GHC-1629/index.md:3]:
```suggestion
summary: Too many type applications to a constructor in a pattern.
```

Move to a passive tone to match the rest of the examples and be more specific that these are type applications.

File: message-index/messages/GHC-1629/index.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+severity: error
+introduced: 9.6.1
+---
+
+Every type application has to be "consumed" by a type variable in the
+corresponding type.  It is an error to apply more types than there are
```
Comment (david-christiansen) [inline message-index/messages/GHC-1629/index.md:9]:
```suggestion
corresponding type.  Just as it is an error to pass more than one argument to `putStrLn`,  it is also an error to apply more types than there are
```

File: message-index/messages/GHC-1629/index.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+type variables in a type.
```
Comment (PPKFS) [inline message-index/messages/GHC-1629/index.md:10]:
```suggestion
type variables in a type. For instance, if a type has only 1 type parameter (e.g. `Maybe a`), we can only apply 1 type argument (e.g. applying `@Int` to produce `Maybe Int`); there is no unbound type parameter present for further applications.
```

**Review by david-christiansen — CHANGES_REQUESTED** [review]
I think it's basically fine - I'm not good at harshness, though :)

**Comment by fisx** [comment]
thanks for the feedback!  :)

File: message-index/messages/GHC-1629/example1/index.md (hunk 2 above)
Comment (fisx) [reply message-index/messages/GHC-1629/example1/index.md:4]: [6441dd6](https://github.com/haskell/error-messages/pull/123/commits/6441dd6dbcd90920b1a25830921a39c16eedc479)

**Review by PPKFS — APPROVED** [review]

**Review by david-christiansen — APPROVED** [review]

**Comment by david-christiansen** [comment]
Doesn't GHC say `GHC-01629`?

**Pull request merged**